pkg/costmodel/costmodel.go

```diff
@@ -2358,6 +2327,68 @@ func getStatefulSetsOfPod(pod v1.Pod) []string {
 	return []string{}
 }
 
+// getGPUCount leverages the k8s API to identify the number of GPUs and vGPUs on
+// the node.
+func getGPUCount(cache clustercache.ClusterCache, n *v1.Node) (*float64, *float64, error) {
```

Comment:
nitpicky, and an optional suggestion - pointers to primitives aren't my favorite. To me reading this with limited familiarity, what does a nil GPU count mean? an error in processing? something with no GPU overrides? 

I would return a negative number here if nothing could be found - that clearly identifies it as bogus and can be discarded. 

Again, personal opinion here and open to push back

Reply:
In this case a nil GPU count means that when parsing the node's metadata/labels we could not find a GPU count, which is valid if the node is not equipped with a GPU to begin with.

I'll go ahead and update the comments here above the function signature to make that clear. I'm also open to returning `-1` as opposed to `nil`, whichever you think is more clear and better practice. Lmk what you think still!

Reply:
Ok I've updated so that the function returns `-1` as opposed to `nil` when no GPU could be identified. I've also updated the function comments! Lmk what you think.